system/src/system_setup.cpp

```diff
@@ -317,12 +317,17 @@ template<typename Config> void SystemSetupConsole<Config>::handle(char c)
     else if ('m' == c)
     {
         print("Your device MAC address is\r\n");
+    #if !HAL_PLATFORM_WIFI    
         IPConfig config = {};
         auto conf = static_cast<const IPConfig*>(network_config(0, 0, 0));
         if (conf && conf->size) {
             memcpy(&config, conf, conf->size);
         }
         const uint8_t* addr = config.nw.uaMacAddr;
+    #else
+        auto conf = static_cast<const IPConfig*>(network_config(NETWORK_INTERFACE_WIFI_STA, 0, 0));
```

Comment:
You still need to check `conf && conf->size` for this case. Why not:
```cpp
        IPConfig config = {};
#if !HAL_PLATFORM_WIFI
        auto conf = static_cast<const IPConfig*>(network_config(0, 0, 0));
#else
        auto conf = static_cast<const IPConfig*>(network_config(NETWORK_INTERFACE_WIFI_STA, 0, 0));
#endif // HAL_PLATFORM_WIFI
        if (conf && conf->size) {
            memcpy(&config, conf, conf->size);
        }
        const uint8_t* addr = conf->nw.uaMacAddr;
```

Reply:
Fixed.